wrangler-core/src/main/java/io/cdap/wrangler/utils/ArtifactSummaryComparator.java

```diff
@@ -22,34 +22,38 @@
 import io.cdap.cdap.api.artifact.ArtifactVersion;
 
 /**
- * Artifact selector util
+ * Comparator for artifact summary
  */
-public class ArtifactSelectorUtils {
-  private ArtifactSelectorUtils() {
-  }
+public class ArtifactSummaryComparator implements Comparable<ArtifactSummary> {
+  private final ArtifactVersion summaryVersion;
+  private final ArtifactSummary summary;
 
-  /**
-   * Pick up the latest artifact, this method assumes the name of the artifact is same
-   */
-  public static ArtifactSummary pickLatest(ArtifactSummary artifact1, ArtifactSummary artifact2) {
-    return isLeftArtifactLarger(artifact1, artifact2) ? artifact1 : artifact2;
+  public ArtifactSummaryComparator(ArtifactSummary summary) {
+    this.summaryVersion = new ArtifactVersion(summary.getVersion());
+    this.summary = summary;
   }
 
-  /**
-   * Check if the first artifct is more recent than second artifact
-   */
-  public static boolean isLeftArtifactLarger(ArtifactSummary artifact1, ArtifactSummary artifact2) {
+  @Override
+  public int compareTo(ArtifactSummary other) {
     // first compare the artifact
-    int cmp = new ArtifactVersion(artifact1.getVersion()).compareTo(new ArtifactVersion(artifact2.getVersion()));
+    int cmp = summaryVersion.compareTo(new ArtifactVersion(other.getVersion()));
     if (cmp > 0) {
-      return true;
+      return 1;
     }
 
     if (cmp < 0) {
-      return false;
+      return -1;
     }
 
     // if scope is different, whoever has user scope is latest
-    return artifact1.getScope().equals(ArtifactScope.USER);
+    return summary.getScope().equals(ArtifactScope.USER) ? 1 : -1;
+  }
+
+  /**
+   * Pick up the latest artifact, this method assumes the name of the artifact is same
+   */
+  public static ArtifactSummary pickLatest(ArtifactSummary artifact1, ArtifactSummary artifact2) {
+    ArtifactSummaryComparator comparator = new ArtifactSummaryComparator(artifact1);
```

Comment:
it's better not to create objects needlessly like this. You can have a static instance of a Comparator and use that here instead of always creating a new object.